catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -1104,4 +1104,14 @@ public static boolean isUserLoggedIn(Context context) {
 	public static String getNumberStringForBricks(float value) {
 		return (int) value == value ? "" + (int) value : "" + value;
 	}
+
+	public static <T extends Object> List<T> distinctListByClassOfObjects(List<T> listToDistinct) {
+		List<T> distinctList = new ArrayList<>();
+		HashMap<Class, T> uniqueMap = new HashMap<>();
+		for (T objectInstance : listToDistinct) {
+			uniqueMap.put(objectInstance.getClass(), objectInstance);
+		}
+		distinctList.addAll(uniqueMap.values());
+		return distinctList;
```

Comment:
> Map instead of Hashmap on a local variable? thats not picky, it's useless. But here you go.

* Program to interfaces, not implementations
* `Vector v = new Vector()` implies that the functionality of Vector is needed (e.g. a function which the Vector has only or thread-safety).
* Dependency inversion principle
* Liskov substitution principle
* Effective Java Item 52
  * Refer to objects by their interfaces
  * > If you get into the **habit** of using interfaces as types, …

I know your original code and that you were using implementations as [return types](https://github.com/krellsebastian/Catroid/commit/44742bd64d10586e8b2a8d0a35d61d565cc74012#diff-addd92b36328ca27d9d2ede2ca3ab9c5R915), [parameters](https://github.com/krellsebastian/Catroid/commit/44742bd64d10586e8b2a8d0a35d61d565cc74012#diff-0b1215285c675bebd81342f589acb64eR57), [members](https://github.com/krellsebastian/Catroid/commit/44742bd64d10586e8b2a8d0a35d61d565cc74012#diff-0b1215285c675bebd81342f589acb64eR43) and I don't even know about [this cast](https://github.com/krellsebastian/Catroid/commit/44742bd64d10586e8b2a8d0a35d61d565cc74012#diff-69b6ec2fdde36c910b08ab1eafb094abR600). Maybe you're coming from a dynamic typed language or you just aren't used to interfaces. That's no problem. Just start to break your bad habit by using interfaces all the way down to local variables.

And always feel free to ask. (But don't expect an answer too sone because currently I'm really short on time.)